Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,522 @@
+bazel build :MyProject
+```
+
+The rule definition for the `xcode_project` may be found in the [github
+repository](https://github.com/pinterest/xchammer/blob/master/BazelExtensions/xcodeproject.bzl).
+Simply put, the aspect traverses sources, and invokes the `xchammer` binary with
+a JSON file. Internally, XCHammer instantiates
+[XcodeGen](https://github.com/yonaskolb/XcodeGen) types and writes them out to
+disk with [xcodeproj](https://github.com/tuist/XcodeProj).
+
+Generating an Xcode project with XCHammer:
+```
+command line -> bazel -> rule -> xchammer -> xcodegen -> project.xcodeproj
+```
+
+Generators like XCHammer and Tulsi take care of integrating Bazel into the IDE.
+Bazel builds are invoked a shell script build phase from the IDE. Basically,
+Xcode shells out to Bazel to produce the application, and then Xcode picks up
+the product from the derived data path. Performing a Bazel build from Xcode.
+
+```
+play button -> shell script build phase -> bazel build ios-app
+```
+
+![Docs](XcodeBazelRunscript.png)
+
+_In the Xcode world, Xcode's internal build systems produce the application._
+
+## More information
+
+### Gathering information with Bazel query
+
+Bazel query is command line interface that uses a DSL designed to query the
+build graph and extract information. This can be used for tooling purposes, or
+to simply questions about the build. Here's a few examples:
+
+What extensions does the target `pinterest` depend upon?
+```
+bazel query 'kind(ios_extension, deps(pinterest))'
+```
+
+What are the labels of source files in `app-lib`?
+```
+bazel query --noimplicit_deps 'labels(srcs, app-lib)'
+```
+
+Please see the [Bazel query
+documentation](https://docs.bazel.build/versions/master/query.html) for more
+information.
+
+### Fixing common Bazel errors
+
+This segment indicates how to fix a basic Bazel error.
+
+Bazel has a few levels of validation which often occur at BUILD time. Note that
+since Bazel doesn't type check all files in the WORKSPACE on every build it's
+possible to have an error in one target, while other targets still work.
+
+```
+ERROR: /path/to/myproject/BUILD:4:13:
+objc_library() got unexpected keyword argument: copts
+```
+
+In the hypothetical file `/path/to/myproject/BUILD`, at the line `4:13` got an
+unexpected keyword argument, `copts` to the macro `objc_library`. The erroneous
+code resides in the example in the [macros](#macros) segment, which calls a
+custom `objc_library` macro. The macro only exposes the parameters `name`,
+`srcs`, `hdrs`, `deps`, and `data`. By attempting to pass `copts` and deviate
+from the default `copts`, Bazel triggered an error.
+
+If needed, Bazel generally will indicate the `.bzl` file where the issue
+occurred and `BUILD` file that created the error in a call stack like fashion.
+
+To actually fix this error, simply remove the unsupported argument `copts`.
+
+ _note: Bazel's diagnostic
+convention is similar to how clang and Swift errors look and feel inside of
+Xcode_ 
```

Comment:
```suggestion
 _Note: Bazel's diagnostic
convention is similar to how clang and Swift errors look and feel inside of
Xcode._ 
```

I suspect most new Bazel users won't agree because fix-its and inline messages are a big component of the "feel"